crates/ruff_python_formatter/src/statement/stmt_import_from.rs

```diff
@@ -43,16 +43,29 @@ impl FormatNodeRule<StmtImportFrom> for FormatStmtImportFrom {
             }
         }
 
-        let comments = f.context().comments().clone();
-        let dangling_comments = comments.dangling_comments(item.as_any_node_ref());
-        write!(f, [trailing_comments(dangling_comments)])?;
-
         let names = format_with(|f| {
             f.join_comma_separated(item.end())
                 .entries(names.iter().map(|name| (name, name.format())))
                 .finish()
         });
-        parenthesize_if_expands(&names).fmt(f)
+
+        // A dangling comment on an import is a parenthesized comment, like:
+        // ```python
+        // from example import (  # comment
+        //     A,
+        //     B,
+        // )
+        // ```
+        let comments = f.context().comments().clone();
+        let parenthesized_comments = comments.dangling_comments(item.as_any_node_ref());
+
+        if parenthesized_comments.is_empty() {
+            parenthesize_if_expands(&names).fmt(f)
+        } else {
+            parenthesized("(", &names, ")")
+                .with_dangling_comments(parenthesized_comments)
+                .fmt(f)
```

Comment:
I'm also changing the `ImportFrom` handling here to preserve:

```python
from example import (  # comment
    A,
    B,
)
```

Previously, we flattened this.